### PR TITLE
oneVideo Adapter - Content Object support (SAPR-15477)

### PR DESCRIPTION
**Pull request by adam-browning**
## Type of change

- [ x] Feature

## Description of change

The oneVideoBidAdapter now supports the passing of OpenRTB imp[0].content Object

You can now pass the Content Object using `bid.params.video.content = {}`
Not all parameters are supported, this is what the VSSP ad-server currently supports:
- id: string
- title: string
- series: string
- season: string
- episode: Number
- cat: Array of strings
- genre: string
- contentrating: string
- language: string 2 letter country code all caps
- prodq: Number
- context: Number
- livestream: Number
- len: Number
- ext: {network: string, channel: string}

See https://www.iab.com/wp-content/uploads/2016/03/OpenRTB-API-Specification-Version-2-5-FINAL.pdf for more details on expected values

- test parameters for validating bids
**Note: This is using e2etest mode**
```
        const adUnits = [{
            code: 'video1',
            mediaTypes: {
                video: {
                    context: 'outstream',
                    playerSize: [640, 480]
                }
            },
            bids: [{
                bidder: 'oneVideo',
                params: {
                    video: {
                        playerWidth: 640,
                        playerHeight: 480,
                        mimes: ['video/mp4', 'application/javascript'],
                        protocols: [2, 5],
                        api: [1, 2],
                        e2etest: true,
                        content: {
                            id: "1234",
                            title: "Title",
                            series: "Series",
                            season: "Season",
                            episode: 1,
                            cat: [
                                "IAB1",
                                "IAB1-1",
                                "IAB1-2",
                                "IAB2",
                                "IAB2-1"
                            ],
                            genre: "Fantase",
                            contentrating: "C-Rating",
                            language: "EN",
                            prodq: 1,
                            context: 1,
                            len: 200,
                            ext: {
                                    network: "ext-network",
                                    channel: "ext-channel"
                                }
                            }
                        },
                        pubId: 'Whatever'
                    }
            }]
        }]
```

## Other information

Hi @kprasadpvv, could you please review?
Thanks in advance,
Adam

### CODE REVIEW

**Comment by mike-chowla** [comment]
@adam-browning Are your changes to package.json intentional?  If not, please revert them.  If there are, please add comment about why they are needed.

File: modules/oneVideoBidAdapter.md [hunk 1]
```diff
@@ -298,3 +298,110 @@ var adUnits = [
...
+                            cat: [
+                                "IAB1",
+                                "IAB1-1",
+                                "IAB1-2",
+                                "IAB2",
+                                "IAB2-1"
+                            ],
+                            genre: "Genre",
+                            album: "Album",
+                            isrc: "Isrc",
```
Comment (kprasadpvv) [inline modules/oneVideoBidAdapter.md:339]: `contentStringKeys` array @ line 323 in adapter doesn't have many strings like `iSrc` , `producer`, `url`... is it intended?

File: modules/oneVideoBidAdapter.md [hunk 2]
```diff
@@ -298,3 +298,89 @@ var adUnits = [
...
+                        mimes: ['video/mp4', 'application/javascript'],
+                        protocols: [2, 5],
+                        api: [1, 2],
+                        ttl: 300,
+                        content: {
+                            id: "1234",
+                            title: "Title",
+                            series: "Series",
+                            season: "Season",
+                            episode: "Episode",
```
Comment (kprasadpvv) [inline modules/oneVideoBidAdapter.md:328]: As per adapter `episode` is not in string as it is in `contentNumberkeys`

File: modules/oneVideoBidAdapter.md (hunk 1 above)
Comment (adam-browning) [reply modules/oneVideoBidAdapter.md]: Hi @kprasadpvv, thanks for catching.
In my first iteration I supported all ORTB 2.5 content object params but after validating with Vinay I found that the ad-server does not support all of them so I reduced the functionality to match.
I have removed the redundant params from the example accordingly.
Thank you Sir

File: modules/oneVideoBidAdapter.md (hunk 2 above)
Comment (adam-browning) [reply modules/oneVideoBidAdapter.md]: fixed

**Comment by adam-browning** [comment]
Hello @mike-chowla, thank you for flagging that, I have reverted the package and package-lock files in this PR accordingly.
FYI - the circleci failing is unrelated to this PR or the oneVideoBidAdapter.
Thank you Sir

**Review by kprasadpvv — APPROVED** [review]

**Comment by musikele** [comment]
Hey, I'm trying to run exactly the example I see in this PR but I get this error in browser's console: 

```
Prebid WARNING: Invalid bid sent to bidder oneVideo: {"bidder":"oneVideo","params":{"video":{"playerWidth":640,"playerHeight":480,"mimes":["video/mp4","application/javascript"],"protocols":[2,5],"api":[1,2],"e2etest":true,"content":{"id":"1234","title":"Title","series":"Series","season":"Season","episode":1,"cat":["IAB1","IAB1-1","IAB1-2","IAB2","IAB2-1"],"genre":"Genre","contentrating":"C-Rating","language":"EN","prodq":1,"context":1,"len":360,"ext":{"network":"ext-network","channel":"ext-channel"}},"pubId":"Whatever"}},"crumbs":{"pubcid":"b3a5a0dd-9d3e-4b94-ac4e-9944954cd29e"},"fpd":{"context":{"pbAdSlot":"video1"}},"mediaTypes":{"video":{"playerSize":[[640,480]],"context":"instream","mimes":["video/mp4"],"protocols":[1,2,3,4,5,6],"playbackmethod":[2]}},"adUnitCode":"video1","transactionId":"928ae0b7-c7f4-44dd-b533-4dfbf6a5a050","sizes":[[640,480]],"bidId":"324091fa935668","bidderRequestId":"241e43caacf9dc","auctionId":"4921212b-bb3e-4efc-a38a-a78a7efe3fe5","src":"client","bidRequestsCount":1,"bidderRequestsCount":1,"bidderWinsCount":0}
```

I'm using a West Coast ip address and e2etest: true . Is this ok ?

**Comment by adam-browning** [comment]
Hi @musikele, sorry for the delayed response.
I'm not sure why but your JSON config has more settings than needed.
maybe try the following:
1. Switch the context to 'outstream'  - that should invoke the renderer on success.
2. Try removing "crumbs" & "fpd"
3. Remove  "mimes", "protocols" and "playbackmethod" from mediaTypes.video - looks like this is contradicting params.video

like this:
```
{
    "bidder": "oneVideo",
    "params": {
        "video": {
            "playerWidth": 640,
            "playerHeight": 480,
            "mimes": ["video/mp4", "application/javascript"],
            "protocols": [2, 5],
            "api": [1, 2],
            "e2etest": true,
            "content": {
                "id": "1234",
                "title": "Title",
                "series": "Series",
                "season": "Season",
                "episode": 1,
                "cat": ["IAB1", "IAB1-1", "IAB1-2", "IAB2", "IAB2-1"],
                "genre": "Genre",
                "contentrating": "C-Rating",
                "language": "EN",
                "prodq": 1,
                "context": 1,
                "len": 360,
                "ext": {
                    "network": "ext-network",
                    "channel": "ext-channel"
                }
            },
            "pubId": "Whatever"
        }
    },
    "mediaTypes": {
        "video": {
            "playerSize": [
                [640, 480]
            ],
            "context": "outstream"
        }
    },
    "adUnitCode": "video1",
    "transactionId": "928ae0b7-c7f4-44dd-b533-4dfbf6a5a050",
    "sizes": [
        [640, 480]
    ],
    "bidId": "324091fa935668",
    "bidderRequestId": "241e43caacf9dc",
    "auctionId": "4921212b-bb3e-4efc-a38a-a78a7efe3fe5",
    "src": "client",
    "bidRequestsCount": 1,
    "bidderRequestsCount": 1,
    "bidderWinsCount": 0
}
```
Let me know if the changes worked
Thanks in advance,
Adam

**Review by musikele — APPROVED** [review]
LGTM , but will wait for CirlceCI fixes

**Comment by adam-browning** [comment]
Hi @musikele I have synced branch with upstream and now it passes test validation.
Should be ready for merging.
Thanks in advance!
Adam

**Pull request merged**